File: algorithms/1_easy/S/02788_split_strings_by_separator/solution.cpp
class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        vector<string> res{};

        for (const string& word : words) {
            istringstream ss{word};
            string part;

            while (std::getline(ss, part, separator)) {
                if (!part.empty()) {
                    res.emplace_back(part);
                }
            }
        }

        return res;
    }
};

File: algorithms/1_easy/S/02788_split_strings_by_separator/solution.rs
impl Solution {
    pub fn split_words_by_separator(words: Vec<String>, separator: char) -> Vec<String> {
        let mut res: Vec<String> = Vec::new();

        for word in words.iter() {
            for part in word.split(separator) {
                if !part.is_empty() {
                    res.push(part.to_string());
                }
            }
        }

        res
    }
}
